src/main/java/org/commonjava/maven/plugins/execroot/AbstractDirectoryGoal.java

```diff
@@ -165,4 +169,28 @@ public void setPluginContext( final Map context )
         this.context = context;
     }
 
+    /**
+     * Resolve all properties of current project placeholder against the new property. 
+     */
+	private void resolveProperties() {
+		Properties projectProperties = currentProject.getProperties();
+
+		for (Enumeration<?> n = projectProperties.propertyNames(); n.hasMoreElements();) {
+			String k = (String) n.nextElement();
+			if (!k.equals(property)) {
+				projectProperties.setProperty(k, getPropertyValue(k, projectProperties));
+			}
+		}
+	}
+
+	private String getPropertyValue(String key, Properties properties) {
```

Comment:
This really needs to use StringSearchInterpolator, so we can handle expressions whose resolved values contain expressions. I'll try to take a look.